### DISCUSSION TITLE
Release 1.17

### DISCUSSION THREAD

**Discussion by webspiderteam** [webspiderteam/Bluetooth-DMM-For-Windows#20] [Announcements]
- Fix Export has null current type and symbol in AUTO mode #19 
- Fix Export has no decimal seperator in some languages #19 
- Hide Titlebar and window buttons when mouse not over
- Chart now in time format
- Flat Icon set to source

<hr /><em>This discussion was created from the release <a href='https://github.com/webspiderteam/Bluetooth-DMM-For-Windows/releases/tag/WPFRelease1.17'>Release 1.17</a>.</em>

**Comment by ludwich66** [comment]
Nice that it continues!

I would like to ask you briefly about the further development:
Do you want to use the iconset?
Do you want to change something in the menu structure?
Working in the wiki only makes sense when the "final" shell is in place.

I would be **happy** if you encapsulate the value "ValueF" in the MQTT export as text in a short time - then you can open it in Node red **without JSON errors**. The adjustment as a float is then possible by the user.

Please write me briefly what your plans are?

Greetings Ludwich

Translataed with google!

**Reply by webspiderteam** [reply]
> Do you want to use the iconset?

I included it and u can use it but I'm not sure about using it. I want to ask all users oppinion but there not much right now.



> Do you want to change something in the menu structure?

I don't think so. maybe only a help button that redirects wiki. there is no other think in my mind rigth now.



> the value "ValueF" in the MQTT export as text

there is already ValuS in string format and also u wanted it as float. there is both of them why you want it string again?

**Reply by ludwich66** [reply]
Again for understanding

Corresponding sets
| MQTT | CSV |
--|--
| ValueS & Range   | sValue & Symbol |
|ValueF  & **_exist_ Range != BaseSym _notexist_**|BaseVal & BaseSym|
|Needet  ValueF==BaseVal & **BaseSym**||

MQTT ERROR 
The output as a string avoids a JSON error being generated in Node Red.
That's why I'm so interested in you repairing ValueF.
The adjustment as a float is then possible by the Node Red user

Just for the equivalent export you should adjust that.
The same "payload" for CSV as MQTT. Even luxuriously configurable with MQTT

**I then have no further wishes for your software (except for errors that I find ;-)**

ludwich

**Reply by webspiderteam** [reply]
> The output as a string avoids a JSON error being generated in Node Red.

then it will string not fload or integer value and it wil same as ValueS. althougt I already fixed it cant give error on language change. it will always give value as 11.45 e.g.

there will BaseVal & BaseSym in mqtt just a few minutes later

**Reply by ludwich66** [reply]
> > The output as a string avoids a JSON error being generated in Node Red.
> 
> then it will string not fload or integer value and it wil same as ValueS. althougt I already fixed it cant give error on language change. it will always give value as 11.45 e.g.
> 
> there will BaseVal & BaseSym in mqtt just a few minutes later

Again to avoid misunderstandings
Put the number in single quotes
L

**Reply by webspiderteam** [reply]
now its like that
{
   "Time":"08/10/2022 23:25:38",
   "Device":"Bluetooth DMM",
   "ValueS":"-00,01",
   "Range":"mV",
   "Current":"DC",
   "Symbols":[
      "AUTORANGE"
   ],
   "ValueF":-0.01,
   "BaseVal":-1E-05,
   "BaseSym":"V"
}

**Reply by webspiderteam** [reply]
with single quotes
{ "Time":"08/10/2022 23:25:38", "Device":"Bluetooth DMM", "ValueS":"-00,01", "Range":"mV", "Current":"DC", "Symbols":[ "AUTORANGE" ], "ValueF":'-0.01', "BaseVal":-1E-05, "BaseSym":"V" } 

Fixer/Validator Output
 Info: Replaced incorrect quotes.

**Reply by ludwich66** [reply]
Quick replay ;-)
Pleas send me a Testfile.
L

**Comment by ludwich66** [comment]
I hope JSON Validator like this mix ;-)

**Comment by webspiderteam** [comment]
[Release.zip](https://github.com/webspiderteam/Bluetooth-DMM-For-Windows/files/9303820/Release.zip)
now u can test

**Comment by ludwich66** [comment]
First view looks GooooD!
Now without JSON error.. and my Standard test row, Battery, Resistor, Capacitor, Temp °C/°F, works.

I set up a nightly longtime MQTT measurement with all Values.
Tommorow i compare CSV an MQTT.

Report Tomorrow.

+++ L
